ballerina/tests/test.bal

```diff
@@ -84,6 +85,30 @@ function testCreateObject() {
     }
 }
 
+@test:Config {
+    dependsOn: [testGetObject]
+}
+function testCreatePresignedURLGET() returns error? {
+    log:printInfo("amazonS3Client->createPresignedURL() GET");
+    Client amazonS3Client = check new (amazonS3Config);
+    string response = check amazonS3Client->createPresignedURL(testBucketName, fileName, GET, 3600);
+    http:Client httpClient = check new (response);
+    http:Response httpResponse = check httpClient->get(EMPTY_STRING);
+    test:assertEquals(httpResponse.statusCode, 200, "Failed to create presigned URL");
+}
+
+@test:Config {
+    dependsOn: [testGetObject]
+}
+function testCreatePresignedURLPUT() returns error? {
+    log:printInfo("amazonS3Client->createPresignedURL() PUT");
+    Client amazonS3Client = check new (amazonS3Config);
+    string response = check amazonS3Client->createPresignedURL(testBucketName, fileName, PUT, 3600);
+    http:Client httpClient = check new (response);
+    http:Response httpResponse = check httpClient->put(EMPTY_STRING, content);
+    test:assertEquals(httpResponse.statusCode, 200, "Failed to create presigned URL");
+}
+
```

Comment:
Can't we add error tests?

Reply:
Was [previously suggested](https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/108#discussion_r1530087963) by @NipunaRanasinghe to use check statements instead of of else blocks in order to simplify the code

Reply:
Yeah, but that's for the success scenarios, right? If we have error scenarios, we can assign to a variable that including error and assert. 

In this case, it may not even be `error`, rather an HTTP response with an error status code.

Reply:
```suggestion
function testCreatePresignedUrlPut() returns error? {
    log:printInfo("amazonS3Client->createPresignedUrl() PUT");
    Client amazonS3Client = check new (amazonS3Config);
    string|error url = amazonS3Client->createPresignedURL(testBucketName, fileName, PUT, 3600);
    if url is error {
        test:assertFail(url.toString());
    }
    http:Client httpClient = check new (url);
    http:Response|error httpResponse = httpClient->put(EMPTY_STRING, content);
    if httpResponse is error {
        test:assertFail(httpResponse.toString());
    }
    test:assertEquals(httpResponse.statusCode, 200, "Failed to create presigned URL");
}
```

Reply:
```
function testCreatePresignedUrlPut() returns error? {
    log:printInfo("amazonS3Client->createPresignedUrl() PUT");
    Client amazonS3Client = check new (amazonS3Config);
    string|error url = amazonS3Client->createPresignedUrl(testBucketName, fileName, PUT, 3600);
    if url is error {
        test:assertFail(url.toString());
    }
    http:Client httpClient = check new (url);
    http:Response|error httpResponse = httpClient->put(EMPTY_STRING, content);
    if httpResponse is error {
        test:assertFail(httpResponse.toString());
    }
    test:assertEquals(httpResponse.statusCode, 200, "Failed to create presigned URL");
}
```
Is this what you suggest?

Reply:
No, I meant a negative test for `createPresignedUrl`. So simulating a scenario where `createPresignedUrl` should fail and validating the error.

Reply:
Not sure if I understand
The thing is that createPresignUrl is entirely client side with no interaction with the S3 service APIs, so if given incorrect data, it would generate a URL successfully with the incorrect data.  Then an error only show if the said URL is used.

The function would fail if the arguments are invalid or if empty strings are provided for the arguments, but this hasn't been checked in the other functions.

Reply:
> The function would fail if the arguments are invalid or if empty strings are provided for the arguments, but this hasn't been checked in the other functions.

IMO, we can add a simple test or two for the error scenarios which depend on the input with this function you're adding (even if they may have not been covered for other functions, which IMO, ideally should have been covered).